# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/StencilXdr.cpp
# Commit: ca60758cd534
# Full Hash: ca60758cd534c0f3bb04a7acda15eb41a67e7389
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 12: Use TaggedParserAtomIndex in ObjLiteralStencil. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96919
# ==============================================================================

diff -r 8af30c6d9bb9 -r ca60758cd534 js/src/frontend/StencilXdr.cpp
--- a/js/src/frontend/StencilXdr.cpp	Sat Nov 14 00:03:36 2020 +0000
+++ b/js/src/frontend/StencilXdr.cpp	Sat Nov 14 00:03:45 2020 +0000
@@ -832,9 +832,9 @@
                                               ObjLiteralStencil& stencil) {
   MOZ_TRY(XDRObjLiteralWriter(xdr, stencil.writer_));
 
-  MOZ_TRY(XDRVector(xdr, stencil.atoms_, nullptr));
-  for (const ParserAtom*& entry : stencil.atoms_) {
-    MOZ_TRY(XDRParserAtom(xdr, &entry));
+  MOZ_TRY(XDRVector(xdr, stencil.atoms_));
+  for (auto& entry : stencil.atoms_) {
+    MOZ_TRY(XDRTaggedParserAtomIndex(xdr, &entry));
   }
 
   return Ok();
